interview-questions.md

```diff
@@ -7,55 +7,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a code assigned to an array. You would use them to give value to a set of code that can be called on when needed.
 
-  Researched answer:
+  Researched answer: Its a block of code designed to perform a particular task. You would invoke it when the specific task is needed to be performed.
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: .map() gives you the whole line of code and .filter() gives you specific value thats in the parenthesis
 
-  Researched answer:
+  Researched answer: .map() creates a function that pushes the array through a new sequence of code with the new results done by the .map() function and .filter() runs the array through a check system that searches for the specified condition
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() runs the whole code in the box thats specified while return tells the computer to give you a specified code that goes next to the command.
 
-  Researched answer:
+  Researched answer:  Console.log() writes a message to the console and return ends function execution and specifies a value to the function caller.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: its a line of code that is meant to be ran by the function.
 
-  Researched answer:
+  Researched answer: its an array object accessible inside  a function
```

Comment:
It can be an array or an object, but in the broadest sense it is any data that gets passed into a function.